### ISSUE TITLE
[4.x]: Craft CMS does not support backtracking to this version. Please update to Craft CMS 4.5.4 or later.

### ISSUE THREAD

**Issue by CoolDevMan** [craftcms/cms#13700]
### What happened?

![image](https://github.com/craftcms/cms/assets/54260086/3b1a069e-4b22-4999-a4cc-eea1488ac9e0)

![image](https://github.com/craftcms/cms/assets/54260086/27129291-8aec-42a3-8929-cb5d87f60992)

2 days ago, I updated the CMS version to the latest version on the dashboard screen.
When the update was completed, no problems were found.
The problem seems to have started appearing 11 hours ago.
Currently, the dashboard page cannot be accessed and the API call is failing on the front end as well.
Please let me know the solution.
Thanks in advance.

### Craft CMS version

4.5.5

### PHP version

8.2.10

### Operating system and version

heroku

### Database type and version

mysql

### Installed plugins and versions

-

**Comment by jawadharaba98** [comment]
Look in ur composer.json and change the craft version to the last version and do composer update

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/72a40191-3b85-437f-ad2a-af3672817384)

![image](https://github.com/craftcms/cms/assets/54260086/9825c228-af09-476e-94a0-76e79b110e53)

**Comment by CoolDevMan** [comment]
The cms version of composer.json is 4.5.5, and the composer update has also been completed.
However, the error was not resolved.

**Comment by thupsi** [comment]
Try `rm -rf vendor/ && rm composer.lock && composer clear-cache && composer update`

Do a DB backup first, just in case.

**Comment by CoolDevMan** [comment]
I just ran the above command. The problem is not yet resolved.

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/75443a13-9a6a-464d-a1ac-4fe62ea68fba)

I just discovered one more problem.
After executing all version update commands, when I reconnect to the server and check, the version of craft is still displayed as 4.4.13.
[craft 4.4.13 => 4.5.5 (CRITICAL)]
After executing the cms update command, the above line is not displayed, but if I check it again after some time, it is displayed.
What's the problem?

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/6cbaba88-3d5f-4947-ba08-9a194a7daa5f)

The database backup fails when I try to update the CMS version using the php command. Will this affect the CMS version update?

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/fb842d94-9018-42c6-8d5d-942a284d00b9)

After executing the command as shown in the image, craft will not be displayed in the updateable list.

![image](https://github.com/craftcms/cms/assets/54260086/9b5faa73-c0e0-4dd5-b4dd-db1ce75eae14)

When I reconnect after about 10 minutes and run the php command, craft is displayed in the updateable list.

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/0d2b5cfc-4807-4e4e-aab2-a00963a7a60e)

If I check the schema version of the database, the version is still 4.5.4.

**Comment by CoolDevMan** [comment]
I have a question.

![image](https://github.com/craftcms/cms/assets/54260086/8f1a06cc-5139-4c21-9ef7-c32a548406e1)

$storedSchemaVersion = 4.5.3.0
I don't know how to work this code Craft::$app->schemaVersion
Perhaps Craft::$app->schemaVersion is less than $storedSchemaVersion.
So my dashboard screen looks like this.

![image](https://github.com/craftcms/cms/assets/54260086/fb6b0274-ea6b-484c-8591-bd08fc53c479)

Please share how to properly update craft cms version. Thanks

**Comment by jawadharaba98** [comment]
You could try craft cli by using this command  `craft migrate --type=app`

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/704f4367-a51d-4dd7-b659-0feab76498b4)
The command cannot be executed.

![image](https://github.com/craftcms/cms/assets/54260086/b07889be-580e-4492-8d3b-a05fcc8849ff)

Because our website had stopped working, I changed the version and schemaversion in the database's info table to the old version.
As a result, the website is now operational and the dashboard screen can be accessed.
However, the version of our craft cms is still 4.4.13.
Also, we cannot back up the database from the dashboard screen due to an error.
Please share how to update cms version to 4.5.5 without problem.
Thanks.

**Comment by jawadharaba98** [comment]
`php craft migrate --type=app`

**Comment by CoolDevMan** [comment]
@jawadharaba98 Thank you for the quick reply.
But that doesn't work either.

![image](https://github.com/craftcms/cms/assets/54260086/94ba9cb9-eece-4bfe-ac14-c66bf170c9df)

**Comment by jawadharaba98** [comment]
As the docs say u can also use `php craft update --migrate` to run new database migrations 

https://craftcms.com/docs/4.x/console-commands.html#update

**Comment by CoolDevMan** [comment]
![image](https://github.com/craftcms/cms/assets/54260086/6ba57ee3-754c-4922-b1ea-260154ce7667)

Even if I run this command, the database backup will fail.

The database was downloaded manually using phpmyadmin.
Please share how to update craft cms without problems.

php craft update craft
If I run the above command, it will say that the version update has been completed, but after some time, it will return to its original state and the version will be 4.4.13.
Then, we cannot access the dashboard, the website stops working.

**Comment by brandonkelly** [comment]
@CoolDevMan It sounds like maybe this is a load-balanced environment, and you’re only updating one of the servers, but not all of them. Or the servers are ephemeral and reverting to the previous deployed state after some time.

Look into how the site is hosted, and how it is supposed to be deployed. Typically you should be running Craft updates locally, committing the updated `composer.json` + `composer.lock` + `config/project/*` files to your Git repository, and letting an automated deployment pipeline handle actually pushing the changes to your server(s).

**Comment by CoolDevMan** [comment]
@brandonkelly Thanks for your letting me know.
So, should I change composer.json in the local environment, run composer update, and then upload the changed composer.json + composer.lock + config/project/* files to the server?
What should I do on the server?
Can I run the php craft update craft command?
In fact, the local environment is not currently established.
And the mysql database is on amazonaws.com.

**Comment by CoolDevMan** [comment]
@brandonkelly Thank you.
I changed the composer.json and deleted the composer.lock file and the vendor folder, and ran the composer install command on my locally. After uploading the updated composer.lock file to the Heroku server, I updated the CMS on the Craft Dashboard.
As a result, the Craft dashboard and the website are fully functional.

@brandonkelly @thupsi @jawadharaba98
Thanks to all of you for your advice.
